storage/innobase/sync/srw_lock.cc

```diff
@@ -117,8 +117,13 @@ bool transactional_lock_enabled()
 __attribute__((target("htm"),hot))
 bool xtest()
 {
+# ifdef __s390x__
+  return have_transactional_memory &&
+    __builtin_tx_nesting_depth() > 0;
```

Comment:
Thank you for this fix. I found that the [oldest available GCC documentation that mentions HTM for this platform](https://gcc.gnu.org/onlinedocs/gcc-6.5.0/gcc/S_002f390-and-zSeries-Options.html) was for GCC 6.5. This is also the first documentation where I found a mention of this built-in function.

Because the built-in function was introduced at the same time as the {{htm}} target attribute, I do not think that we need any additional compile-time checks for using this.

Reply:
Based on what I tested on https://godbolt.org, `__builtin_tx_nesting_depth()` is specific to s390x and not applicable to the POWER implementation of HTM (v2.07 Hardware Transactional Memory, not v2.09 Hardware Trace Monitor).

Reply:
Just to be clear, there is no "issue" here, only an observation why the change looks correct.